apps/cowswap-frontend/src/modules/swap/hooks/useTokensBalancesCombined.ts

```diff
@@ -0,0 +1,58 @@
+import { useMemo } from 'react'
+
+import { BalancesState, useTokensBalances } from '@cowprotocol/balances-and-allowances'
+import { useWalletInfo } from '@cowprotocol/wallet'
+
+import { BigNumber } from 'ethers'
+
+import { usePreHookBalanceDiff } from 'modules/hooksStore/hooks/useBalancesDiff'
+
+export function useTokensBalancesCombined() {
```

Comment:
The hook should be out of Swap module, since this is generic entity.
Please, create a new module, like `combinedBalances` and move all related hooks there